packages/office-ui-fabric-react/src/components/Rating/Rating.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+import './Rating.scss';
+import { IRatingProps, RatingSize } from './Rating.Props';
+import { BaseComponent } from '../../common/BaseComponent';
+import { getId } from '../../utilities/object';
+import { css } from '../../utilities/css';
+
+export interface IRatingState {
+  rating: number;
+  focusedRating: number;
+}
+
+export class Rating extends BaseComponent<IRatingProps, IRatingState> {
+  public static defaultProps: IRatingProps = {
+    min: 1,
+    max: 5
+  };
+
+  private _id: string;
+  private _labelId: string;
+
+  constructor(props: IRatingProps) {
+    super(props);
+
+    this.state = {
+      rating: this._getInitialValue(props),
+      focusedRating: null
+    };
+
+    this._id = getId('Rating');
+    this._labelId = getId('RatingLabel');
+  }
+
+  public componentWillReceiveProps(nextProps: IRatingProps) {
+    if (typeof nextProps.rating !== 'undefined' && nextProps.rating !== this.state.rating) {
+      this.setState({
+        rating: this._getClampedRating(nextProps.rating)
+      } as IRatingState);
+    }
+  }
+
+  public render() {
+    let stars: JSX.Element[] = [];
+    for (let i = this.props.min; i <= this.props.max; ++i) {
+      stars.push(this._renderStar(i));
+    }
+
+    return <div className={ css('ms-Rating', this.props.className, {
+      'ms-Rating--large': this.props.size === RatingSize.Large
+    }) } role='application'>
+      <div className={ 'ms-Rating-container' } role='radiogroup'>
+        { stars }
+      </div>
+    </div>;
+  }
+
+  private _renderStar(rating: number): JSX.Element {
+    const inputId = `${this._id}-${rating}`;
+
+    return <div className={ css('ms-Rating-star', {
+      'is-selected': rating <= this.state.rating,
+      'is-inFocus': rating === this.state.focusedRating,
+      'is-disabled': this.props.disabled
+    }) } key={ rating }>
+      <input
+        className='ms-Rating-input'
+        type='radio'
+        name={ this._id }
+        id={ inputId }
+        value={ rating }
+        aria-labelledby={ `${this._labelId}-${rating}` }
+        disabled={ this.props.disabled }
+        checked={ rating === this.state.rating }
+        onChange={ this._onChange.bind(this, rating) }
```

Comment:
Both of these patterns will result in creating new functions everytime the control is re-rendered. I like the way we do it in TextField.tsx constructor.
` this._onChange = this._onChange.bind(this);` 

or define the function as an arrow function to capture `this` context.
`private _onChange = (rating) => {
.....
}`
and use it directly.
`onChange ={ this._onChange }`

@dzearing , do we recommend one of these over another?